homeassistant/components/heos/media_player.py

```diff
@@ -0,0 +1,160 @@
+"""Denon HEOS Media Player."""
+
+from homeassistant.components.media_player import MediaPlayerDevice
+from homeassistant.components.media_player.const import (
+    DOMAIN, MEDIA_TYPE_MUSIC, SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PLAY,
+    SUPPORT_PLAY_MEDIA, SUPPORT_PREVIOUS_TRACK, SUPPORT_STOP,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET, SUPPORT_VOLUME_STEP)
+from homeassistant.const import STATE_OFF, STATE_PAUSED, STATE_PLAYING
+
+from . import DOMAIN as HEOS_DOMAIN
+
+DEPENDENCIES = ['heos']
+
+SUPPORT_HEOS = SUPPORT_PLAY | SUPPORT_STOP | SUPPORT_PAUSE | \
+        SUPPORT_PLAY_MEDIA | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+        SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | SUPPORT_VOLUME_STEP
+
+PLAY_STATE_TO_STATE = {
+    'play': STATE_PLAYING,
+    'pause': STATE_PAUSED,
+    'stop': STATE_OFF
+}
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discover_info=None):
+    """Set up the HEOS platform."""
+    controller = hass.data[HEOS_DOMAIN][DOMAIN]
+    players = controller.get_players()
+    devices = [HeosMediaPlayer(p) for p in players]
+    async_add_devices(devices, True)
+
+
+class HeosMediaPlayer(MediaPlayerDevice):
+    """The HEOS player."""
+
+    def __init__(self, player):
+        """Initialize."""
+        self._player = player
+        self._player.state_change_callback = None
```

Comment:
This line can be removed.

Reply:
I know, but I want for readability all internal variables defined in __init__.

Reply:
`state_change_callback` is a property and is defined by the player class and has a backing field that is initialized to `None`.  This line does nothing, and IMO actually causes confusion because it leads me to think it is doing something and was put there for some reason...  You only need to declare local fields in `__init__`.

Reply:
You are absolutely right, didn't know what I was thinking... =)